pkg/common/util/csr.go

```diff
@@ -0,0 +1,38 @@
+package util
+
+import (
+	"crypto/x509/pkix"
+	"crypto/x509"
+	"crypto/rand"
+	"github.com/spiffe/go-spiffe/uri"
+)
+
+func MakeCSR(privateKey interface{}, spiffeId string) (csr []byte, err error) {
```

Comment:
Should we be opinionated about ecdsa here?

Reply:
thought about keeping it generic since its a util..we call it with a ecdsa privatekey